src/main/java/com/nylas/NeuralCleanConversation.java

```diff
@@ -0,0 +1,53 @@
+package com.nylas;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NeuralCleanConversation extends Message {
+	private String conversation;
+	private String model_version;
+
+	public String getConversation() {
+		return conversation;
+	}
+
+	public void setConversation(String conversation) {
+		this.conversation = conversation;
+	}
+
+	public String getModelVersion() {
+		return model_version;
+	}
+
+	public void setModelVersion(String modelVersion) {
+		this.model_version = modelVersion;
+	}
+
+	public List<File> extractImages(NylasAccount account) throws RequestFailedException, IOException {
+		List<File> fileList = new ArrayList<>();
+		if(conversation != null) {
+			// After applying the regex, if there are IDs found they would be
+			// in the form of => 'cid:xxxx' (including apostrophes), so we discard
+			// everything before and after the file ID (denoted as xxxx above)
```

Comment:
The example with `images_as_markdown` does not use the apostrophes, but I'm not sure if we're trying to support that here.

Reply:
Here's the two forms the images might come in:

https://developer.nylas.com/docs/intelligence/clean-conversations/#inline-images

Reply:
Right, and the second one has no apostrophes.  Do we want to match it here?  Or is this only for the first form?

Reply:
Yeah we'd want to match it in both forms